google/cloud/aiplatform/models.py

```diff
@@ -1227,6 +1198,200 @@ def _instantiate_prediction_client(
             prediction_client=True,
         )
 
+    @staticmethod
+    def _validate_traffic_split(
+        traffic_split: Dict[str, int],
+        endpoint: Optional["Endpoint"] = None,
+        deploy_model: bool = False,
+    ):
+        """Helper method to validate traffic_split against endpoint.
+
+        Args:
+            traffic_split (Dict[str, int]):
+                Required. A map from a DeployedModel's ID to the percentage of
+                this Endpoint's traffic that should be forwarded to that DeployedModel.
+                If a DeployedModel's ID is not listed in this map, then it receives
+                no traffic. The traffic percentage values must add up to 100, or
+                map must be empty if the Endpoint is to not accept any traffic at
+                the moment. Key for model being deployed is "0".
+            endpoint (Endpoint):
+                Optional. The endpoint against which to validate the
+                deployed model ids in the traffic_split.
+                If not set, the endpoint does not exist and there is no deployed model.
+            deploy_model (bool):
+                A boolean value indicate if the traffic_split contains
+                a key for model being deployed, if so, it should be "0". Default to False.
+        Raises:
+            ValueError: if an endpoint is not provided, or any deployed model id within
+                        the traffic_split can not be found at the endpoint
+            ValueError: if sum of all traffic within the traffic_split does not equal 100
+        """
+        traffic_split_model_ids = list(traffic_split.keys())
+        endpoint_model_ids = set(
+            [] if not endpoint else [model.id for model in endpoint.list_models()]
+        )
+
+        if deploy_model:
+            endpoint_model_ids.add(_DEPLOYING_MODEL_TRAFFIC_SPLIT_KEY)
+
+        if any(
```

Comment:
Is the SDK introducing any additional functionality that requires we validate the deployed models locally? If not, then preference to push this validation to the service.

Reply:
Agreed. I'll pick up this branch from Morgan and finish this PR this week.